neuraxle/checkpoints/checkpoint_pipeline_runner.py

```diff
@@ -0,0 +1,100 @@
+from typing import Any
```

Comment:
I'd also suggest restructuring like this to be consistent with the existing structure. Instead of having this file `neuraxle/checkpoints/checkpoint_pipeline_runner.py`, I'd instead have those files: 
- `neuraxle/checkpoints.py` or`neuraxle/checkpoints/*.py` for containing each checkpoint classes
- `neuraxle/runners.py` for containing the current and unique runner. 

This is because the CheckpointPipelineRunner is becoming the default runner, and for this reason, it probably shouldn't be hidden too deep in submodules. The streaming runner doesn't exist yet and it wouldn't be used by default either.